From dfa3ad4ad485141c1bf3040c6efd92bb479bfe0a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Thu, 13 Sep 2018 23:45:29 +0200
Subject: [PATCH] libziparchive: Don't use string_view

This seems to be a C++17 feature [0] and Alpine's GCC doesn't properly
support C++17 yet. The proper way to fix this would of cause be checking
whether or not C++17/string_view is supported.

[0]: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n3921.html
---
 libziparchive/zip_archive.cc | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/libziparchive/zip_archive.cc b/libziparchive/zip_archive.cc
index 5e5e7afd1..409029c2f 100644
--- a/libziparchive/zip_archive.cc
+++ b/libziparchive/zip_archive.cc
@@ -100,10 +100,6 @@ static uint32_t RoundUpPower2(uint32_t val) {
 }
 
 static uint32_t ComputeHash(const ZipString& name) {
-#if !defined(_WIN32)
-  return std::hash<std::string_view>{}(
-      std::string_view(reinterpret_cast<const char*>(name.name), name.name_length));
-#else
   // Remove this code path once the windows compiler knows how to compile the above statement.
   uint32_t hash = 0;
   uint16_t len = name.name_length;
@@ -114,7 +110,6 @@ static uint32_t ComputeHash(const ZipString& name) {
   }
 
   return hash;
-#endif
 }
 
 /*
